x-pack/plugins/security_solution/public/common/components/matrix_histogram/index.tsx

```diff
@@ -209,26 +209,26 @@ export const MatrixHistogramComponent: React.FC<MatrixHistogramComponentProps> =
     () => (title != null && typeof title === 'function' ? title(selectedStackByOption) : title),
     [title, selectedStackByOption]
   );
-  const visualizationResponse = useVisualizationResponse({ visualizationId });
+  const { responses } = useVisualizationResponse({ visualizationId });
   const subtitleWithCounts = useMemo(() => {
     if (isInitialLoading) {
       return null;
     }
 
     if (typeof subtitle === 'function') {
       if (isChartEmbeddablesEnabled) {
-        if (!visualizationResponse || !visualizationResponseHasData(visualizationResponse[0])) {
+        if (!responses || !visualizationResponseHasData(responses[0])) {
```

Comment:
Is it possible for `responses` to equal `[]`? This code will fail if it can. I think it can, `responses` is created from `parseVisualizationData` which is a `reduce` with a starting object of `[]`